apps/camera_pipe/camera_pipe_generator.cpp

```diff
@@ -3,9 +3,178 @@
+        b_gb(x, y) = correction + avg(b_b(x, y), b_b(x+1, y));
+
+        // Now interpolate diagonally to get red at blue and blue at
+        // red. Hold onto your hats; this gets really fancy. We do the
+        // same thing as for interpolating green where we try both
+        // directions (in this case the positive and negative diagonals),
+        // and use the one with the lowest absolute difference. But we
+        // also use the same trick as interpolating red and blue at green
+        // sites - we correct our interpolations using the second
+        // derivative of green at the same sites.
+
+        correction = g_b(x, y)  - avg(g_r(x, y), g_r(x-1, y+1));
+        Expr rp_b  = correction + avg(r_r(x, y), r_r(x-1, y+1));
+        Expr rpd_b = absd(r_r(x, y), r_r(x-1, y+1));
+
+        correction = g_b(x, y)  - avg(g_r(x-1, y), g_r(x, y+1));
+        Expr rn_b  = correction + avg(r_r(x-1, y), r_r(x, y+1));
+        Expr rnd_b = absd(r_r(x-1, y), r_r(x, y+1));
+
+        r_b(x, y)  = select(rpd_b < rnd_b, rp_b, rn_b);
+
+        // Same thing for blue at red
+        correction = g_r(x, y)  - avg(g_b(x, y), g_b(x+1, y-1));
+        Expr bp_r  = correction + avg(b_b(x, y), b_b(x+1, y-1));
+        Expr bpd_r = absd(b_b(x, y), b_b(x+1, y-1));
+
+        correction = g_r(x, y)  - avg(g_b(x+1, y), g_b(x, y-1));
+        Expr bn_r  = correction + avg(b_b(x+1, y), b_b(x, y-1));
+        Expr bnd_r = absd(b_b(x+1, y), b_b(x, y-1));
+
+        b_r(x, y)  =  select(bpd_r < bnd_r, bp_r, bn_r);
+
+        // Resulting color channels
+        Func r, g, b;
+
+        // Interleave the resulting channels
+        r = interleave_y(interleave_x(r_gr, r_r),
+                         interleave_x(r_b, r_gb));
+        g = interleave_y(interleave_x(g_gr, g_r),
+                         interleave_x(g_b, g_gb));
+        b = interleave_y(interleave_x(b_gr, b_r),
+                         interleave_x(b_b, b_gb));
+
+        output(x, y, c) = select(c == 0, r(x, y),
+                                 c == 1, g(x, y),
+                                         b(x, y));
+
+        // These are the stencil stages we want to schedule
+        // separately. Everything else we'll just inline.
+        intermediates.push_back(g_r);
+        intermediates.push_back(g_b);
+    }
+
+    void schedule() {
+        int vec = get_target().natural_vector_size(UInt(16));
+        bool use_hexagon = get_target().features_any_of({Target::HVX_64, Target::HVX_128});
+        if (get_target().has_feature(Target::HVX_64)) {
+            vec = 32;
+        } else if (get_target().has_feature(Target::HVX_128)) {
+            vec = 64;
+        }
+        for (Func f : intermediates) {
+            f.compute_at(intermed_compute_at)
+                .store_at(intermed_store_at)
+                .vectorize(x, 2*vec, TailStrategy::RoundUp)
+                .fold_storage(y, 2);
+        }
+        Func(output).compute_at(output_compute_at)
```

Comment:
Oops, I think that's actually no longer necessary and I left it in when cleaning up old code.